pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -666,14 +676,27 @@ func (p *PriorityQueue) isPodBackingoff(podInfo *framework.QueuedPodInfo) bool {
 
 // SchedulingCycle returns current scheduling cycle.
 func (p *PriorityQueue) SchedulingCycle() int64 {
-	p.lock.RLock()
-	defer p.lock.RUnlock()
+	p.activeQLock.RLock()
+	defer p.activeQLock.RUnlock()
 	return p.schedulingCycle
 }
 
-// determineSchedulingHintForInFlightPod looks at the unschedulable plugins of the given Pod
-// and determines the scheduling hint for this Pod while checking the events that happened during in-flight.
-func (p *PriorityQueue) determineSchedulingHintForInFlightPod(logger klog.Logger, pInfo *framework.QueuedPodInfo) queueingStrategy {
+func (p *PriorityQueue) dumpClusterEvents(inFlightPod *list.Element) []*clusterEvent {
+	var events []*clusterEvent
+	for event := inFlightPod.Next(); event != nil; event = event.Next() {
+		e, ok := event.Value.(*clusterEvent)
+		if !ok {
+			// Must be another in-flight Pod (*v1.Pod). Can be ignored.
+			continue
+		}
+		events = append(events, e)
+	}
+	return events
+}
+
+func (p *PriorityQueue) getClusterEventsForPod(logger klog.Logger, pInfo *framework.QueuedPodInfo) ([]*clusterEvent, queueingStrategy) {
```

Comment:
I'd like to eliminate `queueingStrategy` from this returning value.

Can we check `if len(rejectorPlugins) == 0 {}` at `determineSchedulingHintForInFlightPod` side? If Yes, plus if we can simplify the inside of `if !ok ` as I commented at another place, I belive we can eliminate queueingStrategy.

Reply:
Yes, we can move the check to the `determineSchedulingHintForInFlightPod` directly. If we receive a positive confirmation in the comment above, I will remove `queueingStrategy` from the returning values in `getClusterEventsForPod`.

Reply:
Thanks, let me investigate regarding the removal of `if`s on my side too. 
According to the comments ([ref1](https://github.com/kubernetes/kubernetes/pull/120413/files#r1315165907) [ref2](https://github.com/kubernetes/kubernetes/pull/120413/files#r1315139469)), _maybe_ it's just that those `if`s were necessary to keep some tests pass, and _maybe_ we can just change things on the tests side. Let's see..

Reply:
Okay, now https://github.com/kubernetes/kubernetes/pull/125930 is merged. Let's do this.

Reply:
Done